lib/Differentiator/ReverseModeVisitor.cpp

```diff
@@ -2705,7 +2722,7 @@ Expr* getArraySizeExpr(const ArrayType* AT, ASTContext& context,
     // declarations don't have to be moved to the function global scope.
     bool promoteToFnScope =
         !getCurrentScope()->isFunctionScope() &&
-        m_DiffReq.Mode != DiffMode::reverse_mode_forward_pass;
+        m_DiffReq.Mode != DiffMode::reverse_mode_forward_pass && !keepLocal;
     QualType VDCloneType;
     QualType VDDerivedType;
```

Comment:
```suggestion
    QualType VDDerivedType;
    QualType VDTy = VD->getType();
```

And replace the uses of `VD->getType()` below.

Reply:
Done.